src/Core/Extension.php

```diff
@@ -13,6 +13,8 @@
  * Every object instance gets its own set of extension instances,
  * meaning you can set parameters specific to the "owner instance"
  * in new Extension instances.
+ *
+ * @template T of object
```

Comment:
Through this PR there's a mixture of
`@template T of object`
`@template T`

Are they functionally the same? If so should we just standardise on one (probably `@template T`)?

Reply:
`@template t of object` says "`T` will always be an object, of some kind".
`@template t` (with no `of` part) means "This will be something, but we don't know what yet"

We should be specific when we can.